components/server/src/github/github-repository-provider.ts

```diff
@@ -230,6 +230,77 @@ export class GithubRepositoryProvider implements RepositoryProvider {
         return repos;
     }
 
+    public async searchRepos(user: User, searchString: string): Promise<RepositoryInfo[]> {
+        const logCtx = { userId: user.id, searchString };
+
+        // TODO: look to try graphql api to reduce response payload size
+        // search personal repos
+        const userSearch = this.github.run(user, async (api) => {
+            return api.search.repos({
+                q: `"${searchString}" in:name user:@me sort:updated`,
+                sort: "updated",
+                per_page: 10,
+            });
+        });
+
+        // Attach an error handler to log error and not throw
+        userSearch.catch((err) => {
+            log.warn(logCtx, "Error searching user repos", err);
+        });
+
+        // TODO: Look at using graphql query to grab all orgs w/ some kind of latest activity date
```

Comment:
I plan on doing this before we'd enable the flag, but wanted to start getting feedback sooner. This may be grabbing orgs that aren't necessarily relevant w/ recent activity. I anticipate we'll have similar challenges w/ other scm providers, so can look at it at the same time.